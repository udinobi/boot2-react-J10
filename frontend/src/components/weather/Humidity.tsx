import React from "react"

import styled from "styled-components"

import { SmallerText } from "../lib/StyledText"

export interface Data {
    humidity: number
}

const Humidity = styled(SmallerText)`
    display: inline;
    font-family: "Roboto", "sans-serif";
`

const Icon = styled.i`
    color: #1890FF;
    font-size: 1.4rem;
    padding-right: 8px;
`

export default (props: Data) => 
    <div style={{ marginTop: 20 }}>
        <Icon className="wi wi-humidity" />
        <Humidity>{props.humidity}%</Humidity>
    </div>
