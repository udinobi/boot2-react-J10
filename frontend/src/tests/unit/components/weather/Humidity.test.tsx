import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import Humidity, { Data } from "../../../../components/weather/Humidity"

describe("--> Snapshot for Components > weather > 'Humidity'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            humidity: 0
        }
    })

    it("matches", () => {
        const wrapper = shallow(<Humidity {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
