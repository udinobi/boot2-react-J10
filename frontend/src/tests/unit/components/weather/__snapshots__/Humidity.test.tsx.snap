// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'Humidity' matches 1`] = `
<div
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <styled.i
    className="wi wi-humidity"
  />
  <Styled(Styled(Styled(styled.div)))>
    0
    %
  </Styled(Styled(Styled(styled.div)))>
</div>
`;
